### ISSUE TITLE
iOS TextField, Different behavior strategies on appearing the keyboard (related to focusable TextField)

### ISSUE THREAD

**Issue by mori-atsushi** [JetBrains/compose-multiplatform#3128]
Currently, in iOS, the window will automatically move if the focused element is obscured by the software keyboard.
I would like to disable the behavior of the software keyboard as well as Android.

In Android, the keyboard area can be finely controlled using `WindowInsets`.
In doing so, we disable the default behavior of the software keyboard by calling `WindowCompat.setDecorFitsSystemWindows` and setting `android:windowSoftInputMode` to `adjustResize`.
https://developer.android.com/develop/ui/views/layout/sw-keyboard

I think that there should be the following three modes:
* adjustPan (Move the window, current behavior)
* adjustResize (Resize the window)
* none (Do nothing)

By the way, I am developing a window insets library for Compose Multiplatform.
https://github.com/mori-atsushi/insetsx

Probably related: https://github.com/JetBrains/compose-multiplatform/issues/2595

**Comment by brianguertin** [comment]
This is currently a blocker for switching our iOS apps from UIKit to Compose.

We use a common layout system shared across Android & iOS, and handle all the safe area / keyboard ourselves. Basically, we apply the Android `adjustResize` on both platforms, so we get consistent behavior on both.

This also avoids views being unaccessible on iOS while the keyboard is displayed (due to being panned off-screen).

I couldn't figure out any way to work around this for our app, and instead always end up either with duplicate keyboard margin, or inaccessible views at the top of the scroll area.

![Simulator Screenshot - iPhone 14 - 2023-05-18 at 12 39 12](https://github.com/JetBrains/compose-multiplatform/assets/120002/84694417-4c56-4017-84f2-ba822ed8f949)

There's a reason Android has the "adjustResize" behavior, and I think Compose for iOS will need it as well, to be fully compatible with Compose's "WindowInsets" APIs.

**Comment by brianguertin** [comment]
I noticed @dima-avdeev-jb said they're "working on it right now" regarding `WindowInsets` support for Compose iOS, and I think this is closely related.

https://kotlinlang.slack.com/archives/C0346LWVBJ4/p1682323749331609?thread_ts=1682154592.960459&cid=C0346LWVBJ4

**Comment by brianguertin** [comment]
Another interesting fact, TIL, is that `adjustResize` is effectively the default behaviour in SwiftUI, as of iOS 14, see https://www.fivestars.blog/articles/swiftui-keyboard/

SwiftUI views will shrink to make room for the keyboard unless you specify to ignore they keyboard.

So, just another good reason to support it.

**Comment by dima-avdeev-jb** [comment]
We added a configuration function in this PR https://github.com/JetBrains/compose-multiplatform-core/pull/613:
```
ComposeUIViewController(
    configure = {
        onFocusBehavior = OnFocusBehavior. DoNothing
    }
) {
    // Composable content
}
```
Will be available in the next release (1.5.0)

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3128">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
